File: leetcode/100-999/136. Single Number/solution_4.cpp
class Solution 
{
    public:
        int singleNumber(vector<int>& nums) 
        {
            set<int> numbers(nums.begin(), nums.end());

            int sumComplete = 0;
            for (int n : numbers)
                sumComplete += n * 2;

            int sumCurrent = 0;
            for (int n : nums)
                sumCurrent += n;

            return sumComplete - sumCurrent;
        }
};

File: leetcode/100-999/136. Single Number/solution_4.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var singleNumber = function(nums) 
{   
    const numbers = new Set(nums)

    let sumComplete = 0
    for (const n of numbers)
        sumComplete += n * 2

    let sumCurrent = 0
    for (const n of nums)
        sumCurrent += n

    return sumComplete - sumCurrent
};